MUONMatcher.cxx

```diff
@@ -380,53 +384,99 @@ void MUONMatcher::runEventMatching() {
-            if (mftTrack.getCharge() == gTrack.getCharge())
-              if (MFTlabel[0].getEventID() == event)
-                if (matchingCut(gTrack, mftTrack)) {
-                  gTrack.countCandidate();
-                  if (MFTlabel[0].getTrackID() == MCHlabel[0].getTrackID())
-                    gTrack.setCloseMatch();
-                  auto chi2 = (*mCustomMatchFunc)(gTrack, mftTrack);
-                  if (chi2 < gTrack.getMatchingChi2()) {
-                    gTrack.setBestMFTTrackMatchID(mftTrackID);
-                    gTrack.setMatchingChi2(chi2);
+    if (!mMatchSaveAll) {
+      for (auto &gTrack : mGlobalMuonTracks) {
+        auto MCHlabel = mchTrackLabels.getLabels(GTrackID);
+        if (MCHlabel[0].getEventID() == event) {
+          auto mftTrackID = 0;
+          if (mCustomMatchFunc) { // Custom matching function
+            for (auto mftTrack : mMFTTracks) {
+              auto MFTlabel = mftTrackLabels.getLabels(mftTrackID);
+              if (mftTrack.getCharge() == gTrack.getCharge())
+                if (MFTlabel[0].getEventID() == event)
+                  if (matchingCut(gTrack, mftTrack)) {
+                    gTrack.countCandidate();
+                    if (MFTlabel[0].getTrackID() == MCHlabel[0].getTrackID())
+                      gTrack.setCloseMatch();
+                    auto chi2 = (*mCustomMatchFunc)(gTrack, mftTrack);
+                    if (chi2 < gTrack.getMatchingChi2()) {
+                      gTrack.setBestMFTTrackMatchID(mftTrackID);
+                      gTrack.setMatchingChi2(chi2);
+                    }
                   }
-                }
-            mftTrackID++;
+              mftTrackID++;
+            }
+          } else { // Built-in matching function
+            for (auto mftTrack : mMFTTracks) {
+              auto MFTlabel = mftTrackLabels.getLabels(mftTrackID);
+              if (mftTrack.getCharge() == gTrack.getCharge())
+                if (MFTlabel[0].getEventID() == event)
+                  if (matchingCut(gTrack, mftTrack)) {
+                    gTrack.countCandidate();
+                    if (MFTlabel[0].getTrackID() == MCHlabel[0].getTrackID())
+                      gTrack.setCloseMatch();
+                    auto chi2 = (this->*mMatchFunc)(gTrack, mftTrack);
+                    if (chi2 < gTrack.getMatchingChi2()) {
+                      gTrack.setBestMFTTrackMatchID(mftTrackID);
+                      gTrack.setMatchingChi2(chi2);
+                    }
+                  }
+              mftTrackID++;
+            }
           }
-        } else { // Built-in matching function
+        }
+        GTrackID++;
+      }      // /loop over global tracks
+    } else { // if matchSaveAll is set
+      // store all given matches for performance studies
+      // loop MCH tracks
+      for (auto &track : mMCHTracks) {
+        // printf("BV: MCH track %d \n", GTrackID);
+        mMCHTrackExtrap.extrapToVertexWithoutBranson(&track, mMatchingPlaneZ);
```

Comment:
Because the Branson correction is fixing the final point of the extrapolation and the goal is not to fix it and let the extrapolation go free to whatever position it gives. The Branson correction is used to fix the final point and calculate (estimate) the final angle, see my presentation from some time ago.

Reply:
I see that. Perhaps you'd like to have used the Branson Correction here, but called the "without" function?

Reply:
I had a look in the code and for me extrapToMatchingPlane and extrapToVertexWithoutBransonCorrection do exactly the same thing. What are the differences you are talking about?